modules/aioseop_sitemap.php

```diff
@@ -403,7 +446,175 @@ public function __construct() {
 			// TODO is this required for dynamic sitemap?
 			add_action( 'transition_post_status', array( $this, 'update_sitemap_from_posts' ), 10, 3 );
 			add_action( 'after_doing_aioseop_updates', array( $this, 'scan_sitemaps' ) );
+			add_action( 'after_doing_aioseop_updates', array( $this, 'upgrade_excluded_categories_to_excluded_terms' ) );
 			add_action( 'all_admin_notices', array( $this, 'sitemap_notices' ) );
+			add_action( 'admin_enqueue_scripts', array( $this, 'admin_enqueue_scripts' ), 20 );
+			add_action( "wp_ajax_{$this->prefix}", array( $this, 'ajax' ) );
+		}
+
+		/**
+		 * Migrate the values in "Excluded Categories" to "Excluded Terms".
+		 *
+		 * @since 3.0
+		 */
+		function upgrade_excluded_categories_to_excluded_terms() {
+			global $aioseop_options;
+			$aioseop_options = aioseop_get_options();
+			$options   = $aioseop_options['modules'][ "{$this->prefix}options" ];
+
+			if ( ! empty( $options[ "{$this->prefix}excl_categories" ] ) ) {
+				$options[ "{$this->prefix}excl_terms" ] = $options[ "{$this->prefix}excl_categories" ];
+				unset( $options[ "{$this->prefix}excl_categories" ] );
+				$aioseop_options['modules'][ "{$this->prefix}options" ] = $options;
+				update_option( 'aioseop_options', $aioseop_options );
+			}
+		}
+
+
+		/**
+		 * The single entry function to handle all ajax stuff.
+		 *
+		 * @since 3.0
+		 */
+		function ajax() {
```

Comment:
This is a really broad function name. Wouldn't it be better for a more defined name for its purpose? Like, `ajax_terms_search`.

Reply:
it is a single entry function for all ajax requests for this module as the comment indicates.

Reply:
That is not good idea. If more AJAX is added, then this will have to be renamed, or repurposed, but as time progresses, sticking all ajax functionality in 1 function only increases its complexity, and unnecessary processes with unrelated operations. It also helps to use different nonces